GameOfRunes/GameOfRunes/GameEngine/GameEngine.swift

```diff
@@ -12,12 +12,20 @@ import GameplayKit
 class GameEngine {
     private var systemDelegate: SystemDelegate!
     private var removeDelegate: RemoveDelegate!
-    var contactDelegate: ContactDelegate!
+    private (set) var contactDelegate: ContactDelegate!
     private var spawnDelegate: SpawnDelegate!
     private var entities = [EntityType: Set<Entity>]()
     private var toRemoveEntities = Set<Entity>()
     private (set) var metadata: GameMetaData
-    weak var gameScene: GameScene?
+    
+    private (set) weak var gameScene: GameScene?
+    
+    var sceneWidth: CGFloat {
+        gameScene?.size.width ?? CGFloat.zero
```

Comment:
Perhaps use `UIScreen.main.bounds.size.width / height` instead of .zero

Reply:
Think Jeremy has a point :)